### PR TITLE
Fix bug 1426835: Do not render HTML in FTL strings

### PR DESCRIPTION
**Pull request by mathjazz**
@adngdb or @Pike or @jotes If anyone is still around to have a look at this, I'd appreciate it.

### CODE REVIEW

File: pontoon/base/static/js/fluent_interface.js [hunk 1]
```diff
@@ -258,7 +258,13 @@ var Pontoon = (function (my) {
...
         elements.forEach(function (item) {
           if (item.type === 'TextElement') {
-            translatedValue += item.value;
+            // TODO: We shouldn't rely on markPlaceables in determining when to not render HTML.
+            if (markPlaceables) {
+              translatedValue += Pontoon.doNotRender(item.value);
+            }
+            else {
+              translatedValue += item.value;
+            }
```
Comment (mathjazz) [inline pontoon/base/static/js/fluent_interface.js]: This fixes the source string panel.

File: pontoon/base/static/js/fluent_interface.js [hunk 2]
```diff
@@ -601,6 +607,11 @@ var Pontoon = (function (my) {
             var variants = ast.value.elements[0].variants;
             response = this.serializePlaceables(variants[0].value.elements);
           }
+
+          // Update source string markup
+          if (object.hasOwnProperty('original')) {
+            response = Pontoon.doNotRender(response);
+          }
```
Comment (mathjazz) [inline pontoon/base/static/js/fluent_interface.js]: This fixes the string list.

File: pontoon/base/static/js/fluent_interface.js [hunk 3]
```diff
@@ -258,7 +258,13 @@ var Pontoon = (function (my) {
 
         elements.forEach(function (item) {
           if (item.type === 'TextElement') {
-            translatedValue += item.value;
+            // TODO: We shouldn't rely on markPlaceables in determining when to not render HTML.
+            if (markPlaceables) {
+              translatedValue += Pontoon.doNotRender(item.value);
```
Comment (jotes) [inline pontoon/base/static/js/fluent_interface.js]: @mathjazz is there a case when `value` shouldn't be escaped? If I understand correctly strings coming from db should be always escaped via `doNotRender()`

Comment (mathjazz) [reply pontoon/base/static/js/fluent_interface.js]: Yes, we shouldn't escape in editor (textarea).

File: pontoon/base/static/js/fluent_interface.js [hunk 4]
```diff
@@ -258,7 +258,13 @@ var Pontoon = (function (my) {
...
+            }
+            else {
```
Comment (jotes) [inline pontoon/base/static/js/fluent_interface.js]: `} else {`

**Comment by jotes** [comment]
@mathjazz Seems okay-ish for me. I'm always concerned when I see the code that concatenates html-strings in javascript :/ On my localhost this patch fixes the issue with html tags in ftl files.

File: pontoon/base/static/js/fluent_interface.js (hunk 4 above)
Comment (mathjazz) [reply pontoon/base/static/js/fluent_interface.js]: I'm keeping the code style consistent with how @adngdb reformatted the file. I like his code style a lot, but I agree this particular case looks odd.

**Comment by mathjazz** [comment]
Thanks for having a look, @jotes! I'll go ahead and merge since this is pretty nasty.

File: pontoon/base/static/js/fluent_interface.js (hunk 4 above)
Comment (adngdb) [reply pontoon/base/static/js/fluent_interface.js]: Ah, this is another case where it's all about personal taste... I personally dislike `} else {`, I find it harder to identify blocks and logic with it. In general I'm a big fan of vertical space, and every attempt at removing some makes me sad. :)

**Pull request merged**